include/itkTileMerging.h

```diff
@@ -0,0 +1,218 @@
+
+  /** Get/Set background value (used if CropToFill is false).
+   * Default PixelType's value (usually zero) if not set. */
+  itkSetMacro(Background, PixelType);
+  itkGetMacro(Background, PixelType);
+
+  /** CropToFill indicates whether the output image will be cropped so it
+   * entirely consists of input tiles (no default background filling).
+   * If CropToFill is false, the big image will have the extent to include
+   * all of the input tiles. The pixels not covered by any input tile
+   * will have the value specified by the Background member variable. */
+  itkSetMacro(CropToFill, bool);
+  itkGetMacro(CropToFill, bool);
+  itkBooleanMacro(CropToFill);
+
+protected:
+  TileMerging();
+  virtual ~TileMerging() {};
+  void PrintSelf(std::ostream& os, Indent indent) const override;
+
+  /** Method invoked by the pipeline in order to trigger the computation of the registration. */
+  void GenerateData() override;
+
+  /** Method invoked by the pipeline to determine the output information. */
+  void GenerateOutputInformation() override;
+
+  /** Make a DataObject of the correct type to be used as the specified output. */
+  typename DataObject::Pointer MakeOutput(typename Superclass::DataObjectPointerArraySizeType idx) override
+  {
+    return ImageType::New();
+  }
+
+  /** If not already read, reads the image into memory.
+   * Only the part which overlaps output image's requested region is read.
+   * If size of the wantedRegion is zero, only reads metadata. */
+  ImageConstPointer GetImage(TileIndexType nDIndex, RegionType wantedRegion);
+
+  /** A set of linear indices of input tiles which contribute to this region. */
+  using ContributingTiles = std::set<SizeValueType>;
+
+  void SplitRegionAndCopyContributions(
+      std::vector<RegionType>& regions,
+      std::vector<ContributingTiles>& regionContributors,
+      RegionType newRegion,
+      size_t oldRegionIndex,
+      SizeValueType tileIndex);
+
+  /** The region will be inside of the rectangle given by min and max indices.
+   * The min is rounded up, while the max is rounded down. */
+  RegionType ConstructRegion(ContinuousIndexType minIndex, ContinuousIndexType maxIndex);
+
+  /** Calculates distance of index from the closes edge of the region. */
+  SizeValueType DistanceFromEdge(ImageIndexType index, RegionType region);
+
+  /** Resamples a single region into m_SingleImage.
+   * This method does not access other regions,
+   * and can be run in parallel with other indices. */
+  void ResampleSingleRegion(SizeValueType regionIndex);
+
+private:
+    bool      m_CropToFill; //crop to avoid backfround filling?
+    PixelType m_Background; //default background value (not covered by any input tile)
+
+    std::vector<TransformConstPointer> m_Transforms;
+    std::vector<SimpleFastMutexLock>   m_TileReadLocks; //to avoid reading the same tile by more than one thread in parallel
+    std::vector<ImagePointer>          m_Metadata; // images with empty buffers
```

Comment:
Why is this called `Metadata`? What are the empty buffers for? How are they populated?

Reply:
Initially it was just metadata. Later I used them for storing images (possibly just required region). I will rename.